httpclient/httpclient.go

```diff
@@ -62,19 +62,25 @@ type Config struct {
 	Tracer tracer
 	// DialContext allows a dial context to be injected into the HTTP transport.
 	DialContext func(ctx context.Context, network string, addr string) (net.Conn, error)
+	// DisableRateLimitBackoff disables the circuit breaker behaviour when the client receives a 429
+	// response. This is useful if the server rate-limits more granularly than requests-per-client
+	// (e.g. if it uses specific query params or headers when bucketing requests) but should be enabled
+	// with care as it can lead to thrashing the server if not used appropriately.
+	DisableRateLimitBackoff bool
```

Comment:
The other place we have a "disable" feature it's called `NoRetry`, perhaps it makes sense to name this similarly `NoRateLimitBackoff`?

Reply:
https://github.com/circleci/ex/blob/d3725f732b2dfe0df3a23d59721a5aa2851b8cc4/httpclient/httpclient.go#L323